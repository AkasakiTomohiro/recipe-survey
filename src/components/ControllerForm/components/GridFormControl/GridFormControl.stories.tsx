/**
 * @file GridFormControl.stories.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import { GridFormControl } from './GridFormControl';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof GridFormControl> = {
  title : 'components/ControllerForm/GridFormControl',
  render: (args): JSX.Element => <GridFormControl {...args} />
};

export default meta;

type Story = StoryObj<typeof GridFormControl>;

export const Primary: Story = {
  args: {
    children: <div>aaa</div>
  }
};

export const AllGrid: Story = {
  args: {
    children : <div>aaa</div>,
    isAllGird: true
  }
};
