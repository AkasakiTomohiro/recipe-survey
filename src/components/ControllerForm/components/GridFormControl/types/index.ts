/**
 * @file index.ts
 * @author Artan's Projects
 * @copyright © 2024 Artan's Projects. All rights reserved.
 */

export interface IGridFormControlProps {
  
  /**
   * Grid幅を横全体にするか
   */
   allGird?: boolean;
}