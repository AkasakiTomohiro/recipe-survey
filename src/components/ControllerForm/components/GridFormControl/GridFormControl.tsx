/**
 * @file GridFormControl.tsx
 * @author Artan's Projects
 * @copyright Â© 2024 Artan's Projects. All rights reserved.
 */

import FormControl from '@mui/material/FormControl';
import Grid, { GridProps } from '@mui/material/Grid';

import { IGridFormControlProps } from './types';

/**
 * Primary UI component for user interaction
 */
export const GridFormControl = ({ children, isAllGird, ...other }: GridProps & IGridFormControlProps): JSX.Element => {
  return (
    <Grid item xs={12} sm={isAllGird ? 12 : 6} {...other}>
      {children && <FormControl fullWidth variant="standard">{children}</FormControl>}
    </Grid>
  );
};
