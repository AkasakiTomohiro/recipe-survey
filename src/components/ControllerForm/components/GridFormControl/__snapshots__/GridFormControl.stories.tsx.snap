// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ControllerForm/GridFormControl AllGrid smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 css-13i4rnv-MuiGrid-root">
  <div class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root">
    <div>
      aaa
    </div>
  </div>
</div>
`;

exports[`components/ControllerForm/GridFormControl Primary smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-13i4rnv-MuiGrid-root">
  <div class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root">
    <div>
      aaa
    </div>
  </div>
</div>
`;
